otherlibs/stdune/src/string.ml

```diff
@@ -82,6 +82,50 @@ let drop_suffix_if_exists s ~suffix =
   | None -> s
   | Some s -> s
 
+module Caseless = struct
```

Comment:
Perhaps it would make sense to functorize these prefix functions to avoid the copy paste?

Reply:
Yes, that's possible. Another possibility is to add an optional argument to the existing functions. Do you have a preference?

Reply:
This is now done (with a functor).